### DISCUSSION TITLE
Welcome to edupagetesthack Discussions!

### DISCUSSION THREAD

**Discussion by ivanhrabcak** [milemaciciky/edupagetesthack#21] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Shimizaki** [comment]
Zdrávim,

keď chcem použiť "edupagetesthack" tak dostanem správu že test je zabezpečený a odpovede nemusia byť správne. Existuje spôsob ako to obísť ? Predom ďakujem za odpoveď. Pokračujte v tom čo robíte ! Edupage "zlepšováky" sú v dnešnej dobe skutočne viac než užitočné.

**Reply by ivanhrabcak** [reply]
Ahoj, zatial ziaden verejny postup neexistuje:(

**Comment by Jkb12** [comment]
Dobrý deň
Zaujimalo by ma akoto funguje, odkiaľ to získava informácie na edupage, ako sa dajú nájsť aj v kóde edupage alebo tak, veľmi ma to to zaujalo, lebo predtým som sa to snažil nájsť a nejako som neuspel. 
Ďakujem

**Reply by ivanhrabcak** [reply]
Ahoj, všetka logika je [tu](https://github.com/milemaciciky/edupagetesthack/blob/main/src/edu-hack.js)!
Odpovede, (ak k tebe vôbec dostanú) sú iba 'schované' hlboko v stránke (nedajú sa ľahko zistiť iba z DOM-u).

Na získanie odpovedí iba voláme funkciu (`getAllAnswerWidgets`), ktorá vracia všetky widgety (na ktoré sú testy a domáce úlohy rozdelené), kde sú uložené rôzne informácie o otázke a o tom ako ju má prehliadač zobrazovať. Medzi týmito informáciami bývajú aj správne odpovede.

**Reply by Jkb12** [reply]
Ďakujem veľmi pekne, už tomu aspoň chápem. :)
Veľky rešpekt že ste boli ochotní to robiť a podarilo sa.

**Comment by sebbery** [comment]
Čauko , funguje to stále ? Poprípade ako vytvorím rozšírenie ?

**Reply by ivanhrabcak** [reply]
Ahoj, rozšírenie už pravdepodobne nefunguje, edupage už neposiela správne odpovede pri žiadnom type materiálu. Ak si chceš spraviť vlastné rozšírenie, budeš musieť nájsť inú metódu ako získať data testu, z endpointu kde edupage stále vracia správe odpovede.
